main.tf

```diff
@@ -0,0 +1,105 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = "us-east-1"
+}
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+locals {
+  public_cidr=["10.0.0.0/24", "10.0.1.0/24"]
+  private_cidr=["10.0.2.0/24", "10.0.3.0/24"]
+}
+
+resource "aws_subnet" "Public" {
+  count = 2
+  vpc_id     = aws_vpc.main.id
+  cidr_block = local.public_cidr[count.index]
+
+  tags = {
+    Name = "public${count.index}"
+  }
+}
+
+resource "aws_subnet" "Private" {
+  count = 2
+  vpc_id     = aws_vpc.main.id
```

Comment:
It's a best practice to separate `count` with the rest of arguments using a newline